enhancements/authentication/pod-security-admission.md

```diff
@@ -0,0 +1,113 @@
+see-also:
+replaces:
+superseded-by:
+---
+
+# PodSecurity admission in OpenShift
+
+Kubernetes recently gained PodSecurity policy admission as part of KEP 2579 [1].
+It is a new pod security admission plugin to enforce Kubernetes PodSecurity standards [2].
+It is meant to replace the existing PodSecurityPolicy admission mechanism in Kubernetes.
+
+This enhancement describes the migration path to integrate PodSecurity admission inside OpenShift.
+Special interest is put to design a migration scheme which lets PodSecurity coexist with the existing Security Context Constraints (SCC) mechanism.
+
+[1] https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2579-psp-replacement
+[2] https://kubernetes.io/docs/concepts/security/pod-security-standards/
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Motivation
+
+In contrast to PodSecurityPolicy admission the desire is not to disable the new upstream PodSecurity admission plugin subsystem within OpenShift.
+The new PodSecurity admissions plugin is a validating plugin only. This implies the opportunity to let it coexist with the existing SCC logic.
+
+### Goals
+
+[1] Enable PodSecurity admission
+[2] Design an architecture that lets PodSecurity coexist with SCC logic
+
+### Non-Goals
+
+[1] Deprecate existing SCC API
+
+## Proposal
+
+We propose introducing PodSecurity admission in a multi-step process. We suggest to follow a similar migration scheme as outlined in the PodSecurityPolicy migration recommendation [1]:
+
+[1] https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2579-psp-replacement#podsecuritypolicy-migration
+
+For OpenShift, the following high level migration plan is envisioned:
+
+1. (DONE) Enable the PodSecurity admission plugin in no-op mode but with the ability to audit policy violations.
+
+- https://github.com/openshift/api/pull/1001 add `PodSecurity` to default enabled feature set in OpenShift hyperkube
+- https://github.com/openshift/cluster-kube-apiserver-operator/pull/1217 enable `PodSecurity` admission plugin and provide a default configuration
+- https://github.com/openshift/kubernetes/pull/950 vendor api changes in openshift/kubernetes
+- https://github.com/openshift/cluster-kube-apiserver-operator/pull/1231 enable `PodSecurity` feature gate in kube-apiserver-operator
+
+Initially, PodSecurity if configured with the following settings:
+
+| Policy  | Profile |
+| --------| --------|
+| enforce | privileged |
+| warn    | baseline |
+| audit   | baseline |
+
+This configures PodSecurity to run in "no-op" mode but will emit warnings and audit log events in case of "baseline" violations.
```

Comment:
"enforce: privileged" doesn't sound no-op?

Reply:
it is counter-intuitive, it means "enforce privileged policy rules in this namespace", aka "all workloads are privileged".